couchbase.py

```diff
@@ -0,0 +1,366 @@
+        return None
+
+
+def config(config_values):
+    """
+    Loads information from the Couchbase collectd plugin config file.
+    Args:
+    :param config_values: Object containing config values
+    """
+    global CONFIGS
+
+    plugin_config = {}
+    interval = DEFAULT_INTERVAL
+    collect_mode = DEFAULT_COLLECT_MODE
+    collect_bucket = None
+    username = None
+    password = None
+    api_urls = {}
+
+    required_keys = ('CollectTarget', 'Host', 'Port')
+    opt_keys = ('Interval', 'CollectMode')
+    bucket_specific_keys = ('CollectBucket', 'Username', 'Password')
+
+    for val in config_values.children:
+        if val.key in required_keys:
+            plugin_config[val.key] = val.values[0]
+        # Read optional parameters
+        if val.key in opt_keys and val.key == 'Interval' and val.values[0]:
+            interval = val.values[0]
+        if val.key in opt_keys and val.key == 'CollectMode' and val.values[0]:
+            collect_mode = val.values[0]
+        # Read bucket specific parameters
+        if val.key in bucket_specific_keys and val.key == 'CollectBucket' and \
+                val.values[0]:
+            collect_bucket = val.values[0]
+        if val.key in bucket_specific_keys and val.key == 'Username' and \
+                val.values[0]:
+            username = val.values[0]
+        if val.key in bucket_specific_keys and val.key == 'Password' and \
+                val.values[0]:
+            password = val.values[0]
+
+    # Make sure all required config settings are present, and log them
+    collectd.info("Using config settings:")
+    for key in required_keys:
+        val = plugin_config.get(key)
+        if val is None:
+            raise ValueError("Missing required config setting: %s" % key)
+        collectd.info("%s=%s" % (key, val))
+
+    if plugin_config.get("CollectTarget") == TARGET_BUCKET:
+        if collect_bucket is None:
+            raise ValueError(
+                    "Missing required config setting for bucket CollectBucket")
+        collectd.info("%s=%s" % ('CollectBucket', collect_bucket))
+
+    # Populate the API URLs now that we have the config
+    base_url = ("http://%s:%s/pools" %
+                (plugin_config['Host'], plugin_config['Port']))
+
+    auth = urllib2.HTTPPasswordMgrWithDefaultRealm()
+    if username is None and password is None:
+        username = password = ''
+    collectd.info(
+                "Using username '%s' ans password '%s' " % (
+                    username, password))
+    auth.add_password(None,
+                      user=username,
+                      passwd=password,
+                      uri=base_url)
+    handler = urllib2.HTTPBasicAuthHandler(auth)
+    opener = urllib2.build_opener(handler)
+
+    if plugin_config.get('CollectTarget') == TARGET_NODE:
+        url = "%s/%s" % (base_url, 'default')
+        api_urls[REQUEST_TYPE_NODE] = url
+    elif plugin_config.get('CollectTarget') == TARGET_BUCKET:
```

Comment:
You might want to add a fallback "else" clause in case the user entered something other than NODE or BUCKET